src/guides/components/endpoint.md

```diff
@@ -0,0 +1,253 @@
+# Creating a Custom Endpoint
+
+When developing a plugin, there are two components to consider: the **frontend** and the **backend**.
+
+You can conceptualize the difference between these two components in the context of operating a car.
+
+The frontend includes all the buttons, dials, pedals, and other controls available to you as the driver. Some components, such as sun visors and the rear-view mirror, are purely frontend. However, most frontend components communicate with the "under-the-hood" backend. This relationship is what allows actions like pressing the gas pedal to accelerate the engine - or, in the context of Caido; clicking a button to send a request.
+
+## Frontend
+
+A "_page_" in Caido simply refers to a user interface. For example, the [Replay](https://docs.caido.io/reference/features/testing/replay.html) and [Automate](https://docs.caido.io/reference/features/testing/automate.html) side menu tabs each produce their own page.
+
+Within these pages are components and elements specific to the feature, such as option menus and buttons. It is through these components and elements that the appearance is customized or communication with the backend occurs.
+
+Think of the Caido application as a browser window with multiple open tabs, each corresponding to a specific page.
+
+## Backend
+
+The backend refers to what is available server-side, what is "_under-the-hood_" or "_out-of-sight_". By incorporating a backend aspect to your plugin, you can extend upon the server-side functionality. This includes storing the data produced by your plugin, sending HTTP requests, or processing user-supplied data.
+
+## sdk.api.register()
```

Comment:
Let's use titles that describe the sequence of operations that a user has to do. https://diataxis.fr/how-to-guides/#describe-a-logical-sequence

So the outline for the guide would be something like

- Registering an API Endpoint
- Calling the API Endpoint

`sdk.api.register()` is just one of many functions that needs to be used in order to register an endpoint. No need to give it its own title.